### PR TITLE
Implemented "--unattended" option (Issue #223)

### LINKED ISSUE #223: Add "--unattended" option
It would be awesome if there would be an unattended option for the script like pi-hole has. I'd be willing to implement it if more people are interested. I don't have too much experience, but it wouldn't be too hard to port the functionality from pi-hole to this script.

### PR DESCRIPTION
**Pull request by EWouters**
I ported the functionality from the pi-hole script to fix #223 (#86). I finished implementing it but more people should try it before merging.

- Parameters will be stored in `/etc/pivpn/setupVars.conf`
- added a finalExports function
- added an update_dialogs function to ask what to do if a parameter file was found
- added an updatePiVPN function

Moved some stuff around.

This request is ready for testing.

### CODE REVIEW

File: auto_install/install.sh [hunk 1]
```diff
@@ -1003,33 +1012,102 @@ confOVPN() {
...
+    #setCustomPort
+    # write out the port
+    echo ${PORT} > /tmp/INSTALL_PORT
+    $SUDO cp /tmp/INSTALL_PORT /etc/pivpn/INSTALL_PORT
+    
+    confOpenVPN
+    confNetwork
+    confOVPN
+    
+	# ?? Is this always OK? Also if you only select one DNS server ??
```
Comment (EWouters) [inline auto_install/install.sh]: Is this always OK? Also if you only select one DNS server, i.e. `OVPNDNS2` is empty?

File: auto_install/install.sh [hunk 2]
```diff
@@ -1003,33 +1012,102 @@ confOVPN() {
...
+        echo "DOWNLOAD_DH_PARAM=${DOWNLOAD_DH_PARAM}"
+        echo "PUBLICDNS=${PUBLICDNS}"
+        echo "OVPNDNS1=${OVPNDNS1}"
+        echo "OVPNDNS2=${OVPNDNS2}"
+        echo "SERVER_NAME=${SERVER_NAME}"
+    }>> "${setupVars}"
+}
+
+
+# I suggest replacing some of these names.
```
Comment (EWouters) [inline auto_install/install.sh]: I suggest replacing some of these names. It looks cleaner in the config file.

File: auto_install/install.sh [hunk 3]
```diff
@@ -1003,33 +1012,102 @@ confOVPN() {
...
+finalExports() {
+    # Update variables in setupVars.conf file
+    if [ -e "${setupVars}" ]; then
+        sed -i.update.bak '/pivpnUser/d;/UNATTUPG/d;/pivpnInterface/d;/IPv4dns/d;/IPv4addr/d;/IPv4gw/d;/pivpnProto/d;/PORT/d;/ENCRYPT/d;/DOWNLOAD_DH_PARAM/d;/PUBLICDNS/d;/OVPNDNS1/d;/OVPNDNS2/d;/SERVER_NAME/d;' "${setupVars}"
+    fi
+    {
+        echo "pivpnUser=${pivpnUser}"
+        echo "UNATTUPG=${UNATTUPG}"
+        echo "pivpnInterface=${pivpnInterface}"
+        echo "IPv4dns=${IPv4dns}"
```
Comment (EWouters) [inline auto_install/install.sh]: Should we save `IPv4dns`?

**Comment by EWouters** [comment]
Is there a way to update without generating new keys? I read the wiki about just not updating, but it might be a good time to add this now.

**Comment by EWouters** [comment]
I also added command line options for the pivpn add and pivpn revoke functions. Now I can fully automate the setup and send myself an email with the certificate when everything is installed. :)

**Comment by EWouters** [comment]
Can anyone help me updating the [bash-completion](https://github.com/EWouters/pivpn/blob/test/scripts/bash-completion) file?

**Comment by 0-kaladin** [comment]
@EWouters man I just saw this now.  been really swamped.  Let me review and give you some comments but thanks a ton for this.

**Comment by 0-kaladin** [comment]
@EWouters this is looking good so far.  I've only looked at it online.  Will test on real devices to see if it broke anything somewhere.  But looks good reading it, awesome job.

**Comment by EWouters** [comment]
Great, I'm happy to make any improvements you might suggest. There are also still a few comments in the code where I wasn't sure what was the best approach.

**Comment by troyfontaine** [comment]
Any word on this PR?  Movement seems to have stalled?

**Comment by EWouters** [comment]
Hi @troyfontaine ,

The code was tested and merged into the master, and most of the suggestions and things I was not sure about have since been fixed or become obsolete. Only a few things (like the [bash-completion](https://github.com/EWouters/pivpn/blob/test/scripts/bash-completion) file and the comment on [this](https://github.com/pivpn/pivpn/blob/33e7ef0c9191c86f74d6b7bc06371350d49e3af9/auto_install/install.sh#L1166) line) remain to be fixed/checked, but I don't know how to do that.

Are you referring to anything specific? And have you tried the --unattended option? 

It might be a good idea to check if there were any substantial improvements in the pi-hole installer since this rewrite. I basically ported all the things they did there to this installer.

**Pull request merged**